From 82aafd9a827ef8bb84f8152683f67acdf079fd65 Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@qt.io>
Date: Thu, 16 Oct 2025 14:00:07 +0800
Subject: [PATCH 198/239] Doc: use registerModule in "Use a C++ enum in
 JavaScript" how-to

This is apparently better than using globalObject.

registerModule was introduced in
3464655f5e2adaa1aed8925a9a54b8fb5f238f31.

Pick-to: 6.8
Change-Id: Ic664e65d7e5f8d65004112802e8f19f4f790bb7d
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
(cherry picked from commit cc9ce866093a31067d2fa37f483c59414692a840)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quick/doc/src/qtquick-how-tos.qdoc                   | 4 ++--
 .../auto/quick/doc/how-tos/how-to-cpp-enum-js/script.mjs | 2 ++
 .../how-to-cpp-enum-js/tst_how-to-cpp-enum-js.cpp        | 9 ++++++---
 3 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/src/quick/doc/src/qtquick-how-tos.qdoc b/src/quick/doc/src/qtquick-how-tos.qdoc
index a451b0491d..0db9a0f7f3 100644
--- a/src/quick/doc/src/qtquick-how-tos.qdoc
+++ b/src/quick/doc/src/qtquick-how-tos.qdoc
@@ -81,11 +81,11 @@
 
     To expose a C++ enum to JavaScript (that is, \l QJSEngine, not
     \l QQmlEngine or \l QQmlApplicationEngine), use
-    \l QJSEngine::newQMetaObject():
+    \l {QJSEngine::}{newQMetaObject()} and \l {QJSEngine::}{registerModule()}:
 
     \quotefromfile how-tos/how-to-cpp-enum-js/tst_how-to-cpp-enum-js.cpp
     \skipto QJSEngine engine
-    \printuntil setProperty
+    \printuntil registerModule
     \skipto Backend backend
     \printuntil backend.load()
 
diff --git a/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/script.mjs b/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/script.mjs
index a4f929222a..c9d1cffce9 100644
--- a/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/script.mjs
+++ b/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/script.mjs
@@ -1,6 +1,8 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR GPL-3.0-only
 
+import { Backend } from "MyApp"
+
 //! [file]
 export function backendStatusUpdate(backendStatus) {
     if (backendStatus === Backend.Error) {
diff --git a/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/tst_how-to-cpp-enum-js.cpp b/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/tst_how-to-cpp-enum-js.cpp
index 9519e94518..78c3584a3f 100644
--- a/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/tst_how-to-cpp-enum-js.cpp
+++ b/tests/auto/quick/doc/how-tos/how-to-cpp-enum-js/tst_how-to-cpp-enum-js.cpp
@@ -26,13 +26,16 @@ tst_HowToCppEnumJs::tst_HowToCppEnumJs()
 
 void tst_HowToCppEnumJs::example()
 {
-    QTest::failOnWarning(QRegularExpression(QStringLiteral(".?")));
+    QTest::failOnWarning();
 
     QJSEngine engine;
     engine.installExtensions(QJSEngine::AllExtensions);
 
-    QJSValue backendJsMetaObject = engine.newQMetaObject(&Backend::staticMetaObject);
-    engine.globalObject().setProperty("Backend", backendJsMetaObject);
+    QJSValue metaObjects = engine.newObject();
+    const QJSValue backendJsMetaObject = engine.newQMetaObject(&Backend::staticMetaObject);
+    metaObjects.setProperty("Backend", backendJsMetaObject);
+    // Repeat the two lines above for other types as needed.
+    engine.registerModule("MyApp", metaObjects);
 
     QTest::ignoreMessage(QtDebugMsg, "Backend loaded successfully");
     Backend backend(&engine);
-- 
2.51.2

